pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1210,6 +1210,66 @@ func (configgen *ConfigGeneratorImpl) generateManagementListeners(node *model.Pr
 	return listeners
 }
 
+// buildSidecarVirtualOutboundListener creates the outbound virtual listener
+// which receives traffic for IPTABLES redirect
+func (configgen *ConfigGeneratorImpl) buildSidecarVirtualOutboundListener(env *model.Environment,
+	node *model.Proxy,
+	push *model.PushContext, proxyInstances []*model.ServiceInstance,
+	ipTablesListener *xdsapi.Listener) *xdsapi.Listener {
+
+	hostname := model.Hostname(util.BlackHoleCluster)
+	mesh := env.Mesh
+	redirectPort := &model.Port{
+		Port:     int(mesh.ProxyListenPort),
+		Protocol: model.ProtocolTCP,
+	}
+
+	if len(ipTablesListener.FilterChains) < 1 || len(ipTablesListener.FilterChains[0].Filters) < 1 {
```

Comment:
this is not possible. proto validation will fail. So you can remove this check.

Reply:
Can you clarify why should we rely on proto validation failing instead of making sure the passed input argument is sane?